Cognite.Jetfire.Cli/JetfireClientFactory.cs

```diff
@@ -20,15 +28,42 @@ public static IJetfireClient CreateClient(ISecretsProvider secrets, string clust
         static ICredentials GetCredentials(ISecretsProvider secrets)
         {
             var apiKey = secrets.GetNamedSecret(ApiKeyEnvironmentVariable);
-            if (string.IsNullOrWhiteSpace(apiKey))
+
+            var tokenUrl = secrets.GetNamedSecret(TokenUrlEnvironmentVariable);
+            var clientId = secrets.GetNamedSecret(ClientIdEnvironmentVariable);
+            var clientSecret = secrets.GetNamedSecret(ClientSecretEnvironmentVariable);
+            var tokenScopes = secrets.GetNamedSecret(TokenScopesEnvironmentVariable);
+            var project = secrets.GetNamedSecret(ProjectEnvironmentVariable);
+
+            if (!string.IsNullOrWhiteSpace(apiKey))
             {
-                throw new JetfireCliException($"The {ApiKeyEnvironmentVariable} environment variable must be set");
+                return new ApiKeyCredentials(apiKey);
             }
-            return new ApiKeyCredentials(apiKey);
+            else if (!string.IsNullOrEmpty(tokenUrl) && !string.IsNullOrEmpty(clientId) && !string.IsNullOrEmpty(clientSecret) && !string.IsNullOrEmpty(tokenScopes) && !string.IsNullOrEmpty(project))
```

Comment:
Since you are using Utils, you may consider using the TrimToNull() extension method.